src/components/OrderableList/OrderableList.tsx

```diff
@@ -0,0 +1,259 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+import { useDroppableCollection } from "@react-aria/dnd";
+import { useGrid } from "@react-aria/grid";
+import { ListKeyboardDelegate } from "@react-aria/selection";
+import { mergeProps } from "@react-aria/utils";
+import { Item } from "@react-stately/collections";
+import { useDraggableCollectionState, useDroppableCollectionState } from "@react-stately/dnd";
+import { GridCollection, useGridState } from "@react-stately/grid";
+import { useListState } from "@react-stately/list";
+import { DropTarget, ItemDropTarget } from "@react-types/shared";
+import React, { useRef } from "react";
+import { GridNode } from "@react-types/grid";
+import { InsertionIndicator } from "./InsertionIndicator";
+import { CollectionItem } from "./CollectionItem";
+import { ItemDragState, OrderableListContainerProps, OrderableListItem, OrderableListProps } from "./types";
+import { useMemoizedId } from "@hooks/useMemoizedId";
+
+export const OrderableList = <T extends object>({
+    onMove,
+    items,
+    renderContent,
+    dragDisabled,
+    disableTypeAhead,
+}: OrderableListProps<T>) => (
+    <OrderableListContainer
+        onMove={onMove}
+        renderContent={renderContent}
+        items={items}
+        disableTypeAhead={disableTypeAhead}
+        dragDisabled={dragDisabled}
+    >
+        {(item: OrderableListItem<T>) => (
+            <Item textValue={item.alt}>
+                <></>
+            </Item>
+        )}
+    </OrderableListContainer>
+);
+
+export const OrderableListContainer = <T extends object>(props: OrderableListContainerProps<T>) => {
+    const gridRef = useRef<HTMLDivElement | null>(null);
+
+    const state = useListState(props);
```

Comment:
Please destruct props and explicitly use the necessary ones in useListState

Reply:
done